common/include/serial_port.h

```diff
@@ -2,7 +2,9 @@
 #define SERIAL_PORT_H_
 
 #include <iostream>
+#include <limits>
```

Comment:
If we put a header in a file, please make sure this file is using the definitions in the header. Try to avoid putting some header in a header and then include the later header in a source file.